### PR TITLE
proto 0.30.0 (new formula)

### PR DESCRIPTION
**Pull request by Phault**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Adapted from [the formula for the very similar tool Volta](https://github.com/Homebrew/homebrew-core/blob/master/Formula/v/volta.rb).

### CODE REVIEW

File: Formula/p/proto.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  homepage "https://moonrepo.dev/proto"
+  url "https://github.com/moonrepo/proto/archive/refs/tags/v0.30.0.tar.gz"
+  sha256 "a564e4c8a4d2483b19e02bd1b3cdac5db4ad61cc0e2d217203f0a641897ba7f4"
+  license "MIT"
+  head "https://github.com/moonrepo/proto.git", branch: "master"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
```
Comment (p-linnane) [inline Formula/p/proto.rb:12]: For other maintainers reviewing: looks like this repo uses GitHub pre-releases, so this is required.

File: Formula/p/proto.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+  depends_on "rust" => :build
+
+  on_linux do
+    depends_on "pkg-config" => :build
+  end
```
Comment (p-linnane) [inline Formula/p/proto.rb:18]:
```suggestion
  depends_on "pkg-config" => :build
  depends_on "rust" => :build

  on_linux do
    depends_on "xz"
  end
```

Comment (p-linnane) [reply Formula/p/proto.rb:18]: `peg-config` is in the `rust` dep tree, so we don't need to place it in the conditional.

There's no linkage with `liblzma` on macOS, so we can just set `xz` as a dependent on Linux.

File: Formula/p/proto.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    end
+  end
+
+  def caveats
+    <<~EOS
+      To finish the installation, run:
+        proto setup
+    EOS
+  end
+
```
Comment (p-linnane) [inline Formula/p/proto.rb:38]:
```suggestion
```

Comment (p-linnane) [reply Formula/p/proto.rb:38]: `caveats` are only for things specific to Homebrew packaging, so we should not include this.

**Review by p-linnane — CHANGES_REQUESTED** [review]
Thanks for the PR. I have a few comments & suggestions.

File: Formula/p/proto.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+  test do
+    system bin/"proto", "install", "node", "19.0.1"
+    node = shell_output("#{bin}/proto bin node").chomp
+    assert_match "19.0.1", shell_output("#{node} --version")
+    path = testpath/"test.js"
```
Comment (SMillerDev) [inline Formula/p/proto.rb:43]: Let's add some whitespace between tests
```suggestion
    assert_match "19.0.1", shell_output("#{node} --version")

    path = testpath/"test.js"
```

File: Formula/p/proto.rb (hunk 3 above)
Comment (Phault) [reply Formula/p/proto.rb:38]: `proto setup` is usually called by their install script automatically, it's purpose is to add PROTO_HOME env var, as well as add the shims and bin folders to the PATH.

Alternatively I can replace it with:

```
 To use proto, add the following line (or equivalent) to your shell profile
      e.g. ~/.profile or ~/.zshrc:
        export PROTO_HOME="$HOME/.proto"
        export PATH="$PROTO_HOME/shims:$PROTO_HOME/bin:$PATH"
  Restart your terminal for the settings to take effect.
```

If this doesn't change your opinion, then I'll remove it as requested. :)

File: Formula/p/proto.rb (hunk 2 above)
Comment (Phault) [reply Formula/p/proto.rb]: Thanks for the help here, this was the part I was most confused about.

File: Formula/p/proto.rb (hunk 3 above)
Comment (p-linnane) [reply Formula/p/proto.rb:39]: If the standard install handles this automatically, then I think it's a good reason to have the caveat for Homebrew. Thanks for the explanation, and we can keep it as is.

**Review by p-linnane — APPROVED** [review]
Thanks for being responsive to feedback, and for contributing to Homebrew @Phault!

**Pull request merged**